tests/Circle3d.elm

```diff
@@ -22,6 +25,49 @@ jsonRoundTrips =
         Decode.circle3d
 
 
+throughPoints : Test
+throughPoints =
+    Test.describe "throughPoints"
+        [ Test.fuzz3 Fuzz.point3d
+            Fuzz.point3d
+            Fuzz.point3d
+            "All given points lie on the circle constructed using `throughPoints`"
+            (\p1 p2 p3 ->
+                let
+                    -- if the first three points are collinear it is not possible to construct a circle
+                    -- that passes through them
+                    isValidInput =
+                        Direction3d.from p1 p2
+                            |> Maybe.map
+                                (\direction ->
+                                    Axis3d.with { originPoint = p1, direction = direction }
+                                )
+                            |> Maybe.map
+                                (\axis ->
+                                    Point3d.squaredDistanceFromAxis axis p3 /= 0
+                                )
+                            |> Maybe.withDefault False
+
+                    circle =
+                        Circle3d.throughPoints ( p1, p2, p3 )
+
+                    liesOnCircle point circle =
+                        Point3d.squaredDistanceFrom point (Circle3d.centerPoint circle)
```

Comment:
I think using `Point3d.squaredDistanceFrom` here is a premature optimization - `Point3d.distanceFrom` is much more clear (especially when tolerances are involved) and it's not like we're calling it on a million different points.